chapters/en/chapter9/7.mdx

```diff
@@ -1,214 +1,223 @@
+
+```py
+import gradio as gr
+
+
+def flip_text(x):
+    return x[::-1]
+
+
+demo = gr.Blocks()
+
+with demo:
+    gr.Markdown(
+        """
+    # Flip Text!
+    Start typing below to see the output.
+    """
+    )
+    inp = gr.Textbox(placeholder="Flip this text")
+    out = gr.Textbox()
+
+    inp.change(fn=flip_text, inputs=inp, outputs=out)
+
+demo.launch()
+```
+
+<iframe src="https://hf.space/gradioiframe/course-demos/flip_text/+" frameBorder="0" height="400" title="Gradio app" class="container p-0 flex-grow space-iframe" allow="accelerometer; ambient-light-sensor; autoplay; battery; camera; document-domain; encrypted-media; fullscreen; geolocation; gyroscope; layout-animations; legacy-image-formats; magnetometer; microphone; midi; oversized-images; payment; picture-in-picture; publickey-credentials-get; sync-xhr; usb; vr ; wake-lock; xr-spatial-tracking" sandbox="allow-forms allow-modals allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts allow-downloads"></iframe>
+
+This simple example above introduces 4 concepts that underlie Blocks:
+
+1. Blocks allow you to build web applications that combine markdown, HTML, buttons, and interactive components simply by instantiating objects in Python inside of a `with gradio.Blocks` context. The order in which you instantiate components matters as each element gets rendered into the web app in the order it was created. (More complex layouts are discussed below)
+
+2. You can define regular Python functions anywhere in your code and run them with user input using Blocks. In our example, we have a simple function that "flips" the input text, but you can write any Python function, from a simple calculation to large machine learning model's inference.
+
+3. You can assign events to any Blocks component. This will run your function when the component is clicked/changed/etc. When you assign an event, you pass in three parameters: fn: the function that should be called, inputs: the (list) of input component(s), and outputs: the (list) of output components that should be called.
+
+   In the example above, we run the `flip_text()` function when the value in the Textbox named inp changes. The event reads the value in inp, passes it as the name parameter to `flip_text()`, which then returns a value that gets assigned to our second Textbox named out.
+
+   To see a list of events that each component supports, see the [documentation](https://www.gradio.app/docs/).
+
+4. Blocks automatically figures out whether a component should be interactive (accept user input) or not, based on the event triggers you define. In our example, the first textbox is interactive, since its value is used by the `flip_text()` function. The second textbox is not interactive, since its value is never used as an input. In some cases, you might want to override this, which you can do by passing the appropriate boolean to interactive, a parameter that every component accepts.
```

Comment:
Could you clarify a bit what you mean by "passing the appropriate boolean to interactive"? If "interactive" is an arg, I suggest formatting it as code and providing an explicit example - I didn't seem to see one above